Add an overview page (#1621)
scalar-labs/scalardb: docs/overview.md

# ScalarDB Overview

> ## What is ScalarDB?

Reviewer: What type of serializability and linearizability is supported in ScalarDB is mentioned in `Manages consistency between multiple database`, but I think it might be also good to mention it along with `ACID transactions` in this first section.

Reviewer: @feeblefakie Just remembered I left this. Do you think we don't need to mention "strict serializability" or something here? It might be a bit hard to mention it considering the analytical queries, but I think the strong correctness of ScalarDB for OLTP is one of the strong appeal points.

> ScalarDB is an HTAP engine for diverse databases. It achieves ACID transactions and real-time analytics across databases to simplify the complexity of managing multiple databases. Existing solutions are designed to run only transactions or analytical queries across heterogeneous databases or both on homogeneous databases.

Reviewer 2: Regarding the English expression, I will leave it to @jnmt 
I'd like to suggest constructing a story as a reader, shorten, common words, simple.
```suggestion
ScalarDB is a cross-database HTAP engine using a unique ACID transaction management.
There are several challenges in configuring an existing HTAP database.
- Transactions among different types of databases
- Trade-offs between performance and consistency
ScalarDB provides ACID transactions across multiple databases including non ACID compliant ones.
Analytical queries always answer consistent data without affecting performance.
This feature also relieves microservices developers from complicated designing tasks.
```
I just want to ask to re-arrange the composition, not the contents. Please replace my wrong contents into appropriate ones.

Reviewer 3: > Regarding the English expression, I will leave it to @jnmt

@ochiai-kfj Probably, Josh?? : )

Author: @ochiai-kfj @josh-wong 
Thank you for the suggestions!
I got your ideas, but the suggested ones miss the analytics part (although it says HTAP), so it sounds a bit weird to me.
The suggestion is probably to use bullet points to clearly state the challenges, so let me update the paragraph based on that.

> ScalarDB is an HTAP engine for diverse databases. It achieves ACID transactions and real-time analytics across databases to simplify the complexity of managing multiple databases. Existing solutions are designed to run only transactions or analytical queries across heterogeneous databases or both on homogeneous databases.

Reviewer 3: If we say an HTAP engine here, it might be good to add an analytical flavor in the use cases section since currently nothing related to analytics is mentioned there.

> to run only transactions or analytical queries across heterogeneous databases or both on homogeneous databases.

Also, this part is a bit hard to follow for me. I don't have a good idea, but something like the following? It's a little redundant, though...

Existing solutions are either of the following, although ScalarDB can run both transactional and analytical queries on heterogeneous databases.
- Run transactions only across heterogeneous databases
- Run analytical queries only across heterogeneous databases
- Run both transactional and analytical queries on homogeneous databases

Author: Thank you!
Good point. 
Let me update it based on your suggestions.

> ScalarDB is an HTAP engine for diverse databases. It achieves ACID transactions and real-time analytics across databases to simplify the complexity of managing multiple databases.  Existing solutions are designed to run only transactions or analytical queries across heterogeneous databases or both on homogeneous databases.

Reviewer: ```suggestion
ScalarDB is an HTAP engine for diverse databases. It achieves ACID transactions and real-time analytics across databases to simplify the complexity of managing multiple databases. Existing solutions are designed to run only transactions or analytical queries across heterogeneous databases or both on homogeneous databases.
```

ScalarDB is a versatile solution that currently supports a wide range of databases. This includes major relational databases such as Oracle Database, PostgreSQL, MariaDB, MySQL, Microsoft SQL Server, SQLite, and their compatible databases (e.g., Amazon Aurora, Google AlloyDB, TiDB, YugabyteDB), as well as NoSQL databases like Amazon DynamoDB, Azure Cosmos DB, and Apache Cassandra.
For details on which databases ScalarDB supports, refer to the [Supported Databases](docs/scalardb-supported-databases.md).

## ScalarDB use cases

> ScalarDB can be used in various ways.
> Here are the three primary use cases of ScalarDB.

Reviewer 4: ```suggestion
ScalarDB can be used in various ways. This section describes the following three primary use cases for ScalarDB:

- Managing siloed databases easily
- Managing consistency between multiple database
- Reducing database migration hurdles
```

> ### Manages siloed databases simply

Reviewer 4: ```suggestion
### Managing siloed databases easily

```

> Many enterprises comprise several organizations, departments, and business units to support agile business operations, which leads to siloed information systems; different organizations likely manage different applications with different databases. Managing such siloed databases is challenging because applications must communicate each database separately and properly deal with the differences between databases. ScalarDB simplifies the management of siloed databases with a unified interface, enabling users to treat the databases as if there were a single database.

Reviewer 4: Revised for grammar and sentence flow. In particular, I re-worded the second sentence to avoid overusing the word `different`.

```suggestion
Many enterprises are comprised of several organizations, departments, and business units that support agile business operations, which often leads to siloed information systems. In addition, each organizations often manages their own applications by using different databases. Managing such siloed databases is challenging because applications must communicate with each database separately and properly deal with the differences between each database. 

ScalarDB simplifies managing siloed databases by providing a unified interface, enabling users to treat the databases as if they were a single database.
```

Author: @josh-wong Thanks.
Overall, looking good to me, but my Grammarly suggested this one.
How about this?
```suggestion
Many enterprises comprise several organizations, departments, and business units that support agile business operations, which often leads to siloed information systems. In addition, each organization often manages its own applications by using different databases. Managing such siloed databases is challenging because applications must communicate with each database separately and properly deal with the differences between each database. 
```

Author: @josh-wong 
1. `that support agile business operation` part sounds a little weird to me because I want to say the reasons for having several organizations.
2. `In addition, ` part also sounds a little different from my original intention because this sentence tried to describe more specific meaning of the first sentence.
3. My Grammarly suggests `are composed of` => `comprise` and `each organizations often manages their own applications` => `each organization often manages their its applications`. In both cases, the latter ones look better and correct.

What do you think?

Reviewer 4: > 1. `that support agile business operation` part sounds a little weird to me because I want to say the reasons for having several organizations.

Understood. I'm OK with leaving that part as is.

> 2. `In addition, ` part also sounds a little different from my original intention because this sentence tried to describe more specific meaning of the first sentence, not additional things.

Thank you for clarifying! I've made one minor suggestion in [another comment](https://github.com/scalar-labs/scalardb/pull/1621#discussion_r1558866268).

> 3. My Grammarly suggests `are composed of` => `comprise` and `each organizations often manages their own applications` => `each organization often manages its applications`. In both cases, the latter ones look better and correct.

I'm OK with using `comprise`. `are composed of` is passive voice and, arguably, maybe a bad habit of mine in cases like this😅 For the second one, that's a typo on my part.

> ### Manages consistency between multiple database

Reviewer 4: ```suggestion
### Managing consistency between multiple database

```

> Modern architectures (e.g., microservice architecture) encourage a system to separate a service and its database into smaller subsets to increase system modularity and development efficiency. However, managing separated (different kinds of) databases is challenging because applications must ensure those databases' correct states (i.e., consistencies). ScalarDB simplifies the management of such separated databases with a correctness guarantee (i.e., ACID with strict serializability), enabling users to focus on application development, not guaranteeing consistency between databases.

Reviewer: ```suggestion
Modern architectures (e.g., microservice architecture) encourage a system to separate a service and its database into smaller subsets to increase system modularity and development efficiency. However, managing separated (different kinds of) databases is challenging because applications must ensure those databases' correct states (i.e., consistencies). ScalarDB simplifies the management of such separated databases with a correctness guarantee (i.e., ACID with strict serializability), enabling users to focus on application development, guaranteeing consistency between databases.
```

Author: @komamitsu Thank you!
But, I think `not` is needed because this means that users don't need to guarantee consistency with ScalarDB.

Reviewer: Oh, I see. I misread the intention. It might sound to me a bit confusing...? But I'll rely on @josh-wong about that 😅

Reviewer 4: Although I read this as the intended meaning, I could see how this could be misread. To reduce potential confusion, especially for when we translate this page, how about we change this paragraph as follows?:

```suggestion
Modern architectures, like a microservice architecture, encourage a system to separate a service and its database into smaller subsets to increase system modularity and development efficiency. However, managing disparate databases, especially of different kinds, is challenging because applications must ensure the correct states (or in other words, consistencies) of those databases.

ScalarDB simplifies managing such disparate databases with a correctness guarantee (or in other words, ACID with strict serializability), enabling you to focus on application development without worrying about guaranteeing consistency between databases.
```

> ### Lowers database migration hurdles

Reviewer 4: ```suggestion
### Reducing database migration hurdles

```

> Applications tend to be locked into a database by, for example, using database-specific capabilities. Such database lock-in discourages upgrading or changing the database because it requires rewriting the application. ScalarDB provides a unified interface over diverse databases; thus, once an application is written with ScalarDB, the application becomes portable, which helps achieve database migration without application rewrite.

Reviewer 4: I don't think `once an application is written with ScalarDB` is correct here, but I'm not totally sure if my suggestion is correct. Please confirm🙇🏻‍♂️

I also made a few changes to wording for clarity.

```suggestion
Applications tend to be locked into using a certain database because of the specific capabilities that the database provides. Such database lock-in discourages upgrading or changing the database because doing so requires re-writing the application.

ScalarDB provides a unified interface across diverse databases. Thus, once an application is written to include ScalarDB, the application becomes modular, which helps to achieve seamless database migration without needing to re-write the application.
```

Author: @josh-wong Thanks!
The latter paragraph sounds weird to me.
How about something like this?

```
ScalarDB provides a unified interface across diverse databases. Thus, once an application is written with the ScalarDB interface, the application becomes portable, which helps to achieve seamless database migration without needing to re-write the application.
```

Reviewer 4: I'm OK with that. Thank you!

# ScalarDB Overview

This page describes what ScalarDB is and its primary use cases.

## What is ScalarDB?

> ScalarDB is a cross-database HTAP engine. It achieves ACID transactions and real-time analytics across diverse databases to simplify the complexity of managing multiple databases. Existing solutions are designed to run only transactions or analytical queries across heterogeneous databases, or they run transactions and analytical queries only on homogeneous databases.

Author: Good point.
Let me think about how I can improve the sentences.
But, let's first publish it and crystalize it later since something is better than nothing.

As a versatile solution, ScalarDB supports a range of databases, including:

- Relational databases that support JDBC, such as MariaDB, Microsoft SQL Server, MySQL, Oracle Database, PostgreSQL, SQLite, and their compatible databases, like Amazon Aurora, Google AlloyDB, TiDB, and YugabyteDB.
- NoSQL databases like Amazon DynamoDB, Apache Cassandra, and Azure Cosmos DB.

For details on which databases ScalarDB supports, refer to [Supported Databases](scalardb-supported-databases.md).

## ScalarDB use cases

ScalarDB can be used in various ways. Here are the three primary use cases of ScalarDB.

### Managing siloed databases easily

> Many enterprises comprise several organizations, departments, and business units to support agile business operations, which often leads to siloed information systems; different organizations likely manage different applications with different databases. Managing such siloed databases is challenging because applications must communicate each database separately and properly deal with the differences between databases.

Reviewer 4: Instead of a semicolon (`;`), how about we use `because` to connect these sentences?

```suggestion
Many enterprises comprise several organizations, departments, and business units to support agile business operations, which often leads to siloed information systems because different organizations likely manage different applications with different databases. Managing such siloed databases is challenging because applications must communicate with each database separately and properly deal with the differences between databases.
```

Author: @josh-wong Thanks!
I used `;` to mean more like `specifically` to give concrete examples.
In that case, what would be a good choice?

Reviewer 4: I see. How about one of the following then?:

**Suggestion 1**

```suggestion
Many enterprises comprise several organizations, departments, and business units to support agile business operations, which often leads to siloed information systems; in particular, different organizations likely manage different applications with different databases. Managing such siloed databases is challenging because applications must communicate each database separately and properly deal with the differences between databases.
```

**Suggestion 2**

```suggestion
Many enterprises comprise several organizations, departments, and business units to support agile business operations, which often leads to siloed information systems. A prime example of this is how different organizations likely manage different applications with different databases. Managing such siloed databases is challenging because applications must communicate each database separately and properly deal with the differences between databases.
```

Author: @josh-wong I took the first suggestion. Thank you!
